doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -84,8 +90,62 @@ Currently, SAI already defines a few port attributes to support port auto negoti
 3. If autoneg is enabled and adv_speeds is not configured or empty, SAI must advertise it with all supported speeds.
-4. If autoneg is enabled and adv_interface_types is not configured or empty, SAI must advertise it with all supported interface types.
-5. If autoneg is disabled and interface_type is not configured, SAI must use SAI_PORT_INTERFACE_TYPE_NONE.
+4. If autoneg is enabled and adv_interface_types is not configured or empty, SA
+I must advertise it with all supported interface types.
+5. If autoneg is disabled and interface_type is not configured, SAI must use SA
+I_PORT_INTERFACE_TYPE_NONE.
+6. If autoneg is enabled, the administrative port speed updates should be blocked.
+7. If autoneg is enabled, both the administrative interface type updates and dynamic interface type request from pmon#xcvrd should be blocked.
+8. If autoneg is enabled on a SFP/QSFP/QSFPDD port, SAI should also activate the link-training to dynamically tune the TX FIR.
+9. If autoneg is transitioned from enabled to disabled, the port speed, FEC, interface type and TX FIR should be restored with the corresponding values in the APPL_DB. (e.g. "speed", "fec", "interface_type" and ("preemphasis", "idriver", "ipredriver", "pre1", "pre2", "pre3", "main", "post1", "post2", "post3", "attn") If the corresponding configuration is not available in the APPL_DB, the following SAI driver defaults should be applied to achieve the best backward compatibility.  
+
+  |Lane Count | Default Speed | Default FEC    | Default Medium | Default Interface |
+  |:----------|:--------------|:---------------|:---------------|:------------------|
+  |         8 | 400G          | RS (RS544-2xN) | copper         |               CR8 |
+  |         4 | 100G          | None           | copper         |               CR4 |
+  |         2 | 100G          | RS (RS544-1xN) | copper         |               CR2 |
+  |         4 | 40G           | None           | copper         |               CR4 |
+  |         1 | 25G           | None           | copper         |                CR |
+  |         1 | 10G           | None           | copper         |                CR |
+  |         1 | 1G            | None           | copper         |             SGMII |
+
+10. The FEC mode of the autoneg advertisement should using the following values by default
+
+  | Speed      | Lane Count | Default FEC |
+  |:-----------|:-----------|:------------|
+  | 400G       | 8          | RS          |
+  | 100G       | 4          | RS          |
+  | 100G       | 2          | RS          |
+  | 40G        | 4          | None        |
+  | 25G        | 1          | None        |
+  | 10G        | 1          | None        |
+  |  1G        | 1          | None        |
+
+11. If autoneg is enabled, the administrative port FEC mode updates (i.e. "fec=rs|fc|none" in CONFIG_DB) should be able to alter the FEC mode of the autoneg advertisement, if the configured FEC is not supported by the hardware, it should fallback as follows  
```

Comment:
I don't think this is the intended behavior. What currently we have in SONiC is if FEC is configured along with autoneg enabled, it overrides the auto negotiated value but doesn't change the advertisement.

Reply:
Agree, this is removed

Reply:
Agree, this is removed